src/lang/SourCParser.g4

```diff
@@ -0,0 +1,200 @@
+    | Identifier LeftBracket constant_expression? RightBracket // arrays
+    | Identifier LeftParen parameter_list* RightParen // function prototypes
+    ;
+
+initializer:
+    assignment_expression
+    | LeftBrace initializer_list RightBrace // for struct and arrays initializing
+    ;
+
+initializer_list:
+    initializer
+    | initializer_list Comma initializer
+    ;
+
+
+assignment_expression:
+    conditional_expression // Resolves into tree of operators with identifers/constants as leaf
+    | unary_expression Assign assignment_expression
+    ;
+
+conditional_expression:
+    logical_or_expression
+    | logical_or_expression Question conditional_expression Colon conditional_expression
+    ;
+
+logical_or_expression:
+    logical_and_expression
+    | logical_or_expression OrOr logical_and_expression
+    ;
+
+logical_and_expression:
+    equality_expression
+    | logical_and_expression AndAnd equality_expression
+    ;
+
+equality_expression:
+    relational_expression
+    | equality_expression Equal relational_expression
+    | equality_expression NotEqual relational_expression
+    ;
+
+relational_expression:
+    additive_expression
+    | relational_expression Less additive_expression
+    | relational_expression Greater additive_expression
+    | relational_expression LessEqual additive_expression
+    | relational_expression GreaterEqual additive_expression
+    ;
+
+additive_expression:
+    multiplicative_expression
+    | additive_expression Plus multiplicative_expression
+    | additive_expression Minus multiplicative_expression
+    ;
+
+multiplicative_expression:
+    cast_expression // Resolves into tree of operators involving unary operators like ++,--
+    | multiplicative_expression Star cast_expression
+    | multiplicative_expression Div cast_expression
+    | multiplicative_expression Mod cast_expression
+    ;
+
+cast_expression:
+    unary_expression
+//    | ( <type-name> ) <cast-expression>
+    ;
+
+unary_expression:
+    postfix_expression
+    | PlusPlus unary_expression
+    | MinusMinus unary_expression
+    | unary_operator cast_expression
+    | Sizeof LeftParen type_name RightParen // Simplified to only primitive types
+    ;
+
+unary_operator: And | Star | Plus | Minus | Not;
+
+type_name:
+    Char
+    | Short
+    | Int
+    | Long
+    | Float
+    | Double
+    ;
+
+postfix_expression:
+    primary_expression
+    | postfix_expression LeftBracket expression RightBracket // array access
+    | postfix_expression LeftParen expression* RightParen // function call
+    | postfix_expression Dot Identifier // struct access
+    | postfix_expression Arrow Identifier // struct access
+    | postfix_expression PlusPlus
+    | postfix_expression MinusMinus
+    ;
+
+primary_expression:
+    Identifier
+    | Constant
+    | StringLiteral
+    | LeftParen expression RightParen // unclear what is this since we have function call above
```

Comment:
This should be for allow user to use parenthesis to set operator precedence. But need to check.